patches/extensions-browser-extension_user_script_manager.h.patch

```diff
@@ -0,0 +1,20 @@
+diff --git a/extensions/browser/extension_user_script_manager.h b/extensions/browser/extension_user_script_manager.h
+index 4e3eaeeedfdf970277bef3d3daffcc1698c633c4..297bf07728136fddc95c2a1276d299e6ec32319f 100644
+--- a/extensions/browser/extension_user_script_manager.h
++++ b/extensions/browser/extension_user_script_manager.h
+@@ -19,6 +19,7 @@
+ namespace content {
+ class BrowserContext;
+ }
++class BraveWalletService;
+ 
+ namespace extensions {
+ 
+@@ -35,6 +36,7 @@ class ExtensionUserScriptManager : public ExtensionRegistryObserver {
+   UserScriptLoader* script_loader() { return &loader_; }
+ 
+  private:
++  friend class ::BraveWalletService;
```

Comment:
what is the friend needed for?

Reply:
This has been migrated from the previously existing `extensions/browser/shared_user_script_manager.h` file, now renamed into `extensions/browser/extension_user_script_manager.h` instead (see [this CL upstream](https://chromium-review.googlesource.com/c/chromium/src/+/2639867)), and the reason it's needed is because `BraveWalletService` needs access to the `OnExtensionLoaded()` and `OnExtensionUnloaded()` methods of this class. 

See the commit doing this adaptation here:
https://github.com/brave/brave-core/pull/7866/commits/4af692e0de9878eb8546cf834b3a16a9fe25c0c3

Reply:
not sure why we're using ExtensionUserScriptManager instead of ExtensionRegistryObserver cc @bbondy

Reply:
Btw, this change is no longer happening to `extensions/browser/extension_user_script_manager.h`, but to `extensions/browser/user_script_manager.h` instead after https://github.com/brave/brave-core/pull/7866/commits/0d4882ee41b056c6817719bc642f52de79db728b

Reply:
but it still has the same patch that seems unnecessary if we were using ExtensionRegistryObserver

Reply:
oh wait, it's triggering the load/unload? I need to look at this again

Reply:
Talked to @bridiver on Slack and decided to add a comment in BraveWalletService explaining why this is done this way, see it here: https://github.com/brave/brave-core/pull/7866/commits/2b78a054e82eb012f9b82a59b519eb75b2f07550